# Dataset Summary Report — Student Dataset

### Objective
Use Python and Pandas to load the student CSV dataset and explore its basic structure and characteristics.

### Analysis covered
- Loading the CSV file into a Pandas DataFrame
- Inspecting the first and last few records
- Viewing random samples
- Checking shape, columns, index, and data types
- Using `info()` and `describe()` for dataset inspection
- Identifying missing values
- Summarizing basic observations and findings

## 1. Import Libraries

In [1]:
import pandas as pd
import numpy as np

## 2. Load the CSV Dataset

In [2]:
# Upload Day7_Student_Dataset.csv to Colab, then run this cell.
df = pd.read_csv('Day7_Student_Dataset.csv')

print('Dataset loaded successfully.')
print('Shape:', df.shape)

Dataset loaded successfully. Shape: (25, 7)


## 3. Inspect the First Five Rows

In [3]:
df.head()

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
0,1001,Aarav Sharma,19,Data Science,87.0,91.0,Srinagar
1,1002,Zoya Khan,20,AI,92.0,95.0,Delhi
2,1003,Rohan Mehta,18,Computer Science,76.0,82.0,Mumbai
3,1004,Ananya Singh,21,Data Science,81.0,88.0,Srinagar
4,1005,Kabir Ali,19,AI,69.0,76.0,Lucknow


## 4. Inspect the Last Five Rows

In [4]:
df.tail()

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
20,1021,Aman Kumar,18,Computer Science,74.0,78.0,Delhi
21,1022,Alina Shah,19,Data Science,96.0,98.0,Srinagar
22,1023,Nikhil Sood,22,AI,78.0,83.0,Pune
23,1024,Maryam Khan,20,Computer Science,85.0,88.0,Hyderabad
24,1025,Reyansh Jain,21,Data Science,80.0,84.0,Kolkata


## 5. View Random Samples

In [5]:
# random_state makes the sample reproducible
df.sample(5, random_state=42)

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
8,1009,Arjun Nair,18,Computer Science,84.0,86.0,Kochi
16,1017,Dev Patel,19,AI,86.0,89.0,Ahmedabad
0,1001,Aarav Sharma,19,Data Science,87.0,91.0,Srinagar
23,1024,Maryam Khan,20,Computer Science,85.0,88.0,Hyderabad
11,1012,Aisha Mir,19,Computer Science,68.0,72.0,Delhi


## 6. Check Dataset Shape

In [6]:
print('Number of rows:', df.shape[0])
print('Number of columns:', df.shape[1])

Number of rows: 25
Number of columns: 7


## 7. View Column Names

In [7]:
print(df.columns)

Student_ID
Name
Age
Course
Marks
Attendance_Percentage
City


## 8. Inspect the DataFrame Index

In [8]:
df.index

RangeIndex(start=0, stop=25, step=1)


## 9. Check Data Types

In [9]:
df.dtypes

Student_ID                 int64
Name                      object
Age                        int64
Course                    object
Marks                    float64
Attendance_Percentage    float64
City                      object


## 10. Dataset Information Using `info()`

In [10]:
# info() displays the number of entries, non-null counts, data types, and memory usage.
df.info()

## 11. Descriptive Statistics Using `describe()`

In [11]:
df.describe()

,Student_ID,Age,Marks,Attendance_Percentage
count,25.000000,25.000000,24.000000,24.000000
mean,1013.000000,20.000000,82.250000,87.083333
std,7.359801,1.414214,8.689074,7.252686
min,1001.000000,18.000000,68.000000,72.000000
25%,1007.000000,19.000000,75.500000,82.750000
50%,1013.000000,20.000000,82.500000,88.000000
75%,1019.000000,21.000000,89.250000,92.250000
max,1025.000000,23.000000,96.000000,98.000000


## 12. Descriptive Statistics Including Categorical Columns

In [12]:
df.describe(include='all')

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
count,25.000000,25,25.000000,25,24.000000,24.000000,24
unique,NaN,25,NaN,3,NaN,NaN,13
top,NaN,Aarav Sharma,NaN,Data Science,NaN,NaN,Srinagar
freq,NaN,1,NaN,9,NaN,NaN,6
mean,1013.000000,NaN,20.000000,NaN,82.250000,87.083333,NaN
std,7.359801,NaN,1.414214,NaN,8.689074,7.252686,NaN
min,1001.000000,NaN,18.000000,NaN,68.000000,72.000000,NaN
25%,1007.000000,NaN,19.000000,NaN,75.500000,82.750000,NaN
50%,1013.000000,NaN,20.000000,NaN,82.500000,88.000000,NaN
75%,1019.000000,NaN,21.000000,NaN,89.250000,92.250000,NaN


## 13. Check for Missing Values

In [13]:
# Count missing values in every column
missing_values = df.isnull().sum()
missing_values

,Column,Missing Values
0,Student_ID,0
1,Name,0
2,Age,0
3,Course,0
4,Marks,1
5,Attendance_Percentage,1
6,City,1


## 14. Basic Categorical Feature Summary

In [14]:
print('Course distribution:')
print(df['Course'].value_counts())

print('\nCity distribution:')
print(df['City'].value_counts(dropna=False))

Course distribution:
Course
Data Science        9
AI                  8
Computer Science    8

City distribution:
City
Srinagar      6
Delhi         3
Pune          2
Bengaluru     2
Kolkata       2
Hyderabad     2
Mumbai        1
Lucknow       1
Kochi         1
Jaipur        1
NaN           1
Chandigarh    1
Ahmedabad     1
Chennai       1


## 15. Basic Observations and Findings

1. The dataset contains 25 rows and 7 columns.
2. The available features are Student_ID, Name, Age, Course, Marks, Attendance_Percentage, and City.
3. There are 4 numeric columns (Student_ID, Age, Marks, Attendance_Percentage) and 3 object/categorical columns (Name, Course, City).
4. The DataFrame uses a default RangeIndex from 0 to 24.
5. There is 1 missing value each in Marks, Attendance_Percentage, and City, while the other columns have no missing values.
6. The Marks column has 24 non-missing values, with a mean of 82.25 and a range from 68 to 96.
7. The Attendance_Percentage column has 24 non-missing values, with a mean of 87.08% and a range from 72% to 98%.

## 16. Conclusion

The dataset was successfully loaded and explored using Pandas. The analysis demonstrated how to inspect records, understand the DataFrame structure, examine column names and data types, generate descriptive statistics, and identify missing values. The dataset contains student demographic and academic information, including age, course, marks, attendance, and city. A small amount of missing data is present in three columns and should be handled appropriately before deeper analysis or modeling.